roles/kubespray-defaults/defaults/main.yaml

```
@@ -230,6 +230,9 @@ kube_api_aggregator_routing: false
 # Profiling
 kube_profiling: false
 
+# Whether to deploy the container engine
+deploy_container_engine: inventory_hostname in groups['k8s-cluster'] or etcd_deployment_type != 'host'
```

now that I see it I have a doubt ...
```suggestion
deploy_container_engine: {{ inventory_hostname in groups['k8s-cluster'] or etcd_deployment_type != 'host' }}
```

---

Is it fine to skip this @champtar point?

Anyways we need to replace `k8s-cluster` with `k8s_cluster` because of the commit https://github.com/kubernetes-sigs/kubespray/commit/360aff4a571f4a8ac7603a4a68223e744fad1209

---

I have a problem while upgrading the etcd in #8381 so I trying to debug by printing out the value of `deploy_container_engine`

```yaml
ansible.builtin.debug:
  msg: "deploy_container_engine={{ deploy_container_engine }} host={{ inventory_hostname }}, etcd={{ etcd_deployment_type }}
```

```
TASK [deploy_container_engine]
ok: [callisto-e3] => {
  "msg": "deploy_container_engine=inventory_hostname in groups['k8s_cluster'] or etcd_deployment_type != 'host' host=callisto-e3, etcd=host"
}
```

It's turn out that the variable is literally a string from `kubespray/roles/kubespray-defaults/defaults/main.yaml` so it evaluated as `True` in `{ role: container-engine, tags: "container-engine", when: deploy_container_engine }` and cause the container-engine role to run 


After I apply a fix in #8386 the value is now boolean as expected and the problem solved.

```
TASK [deploy_container_engine] 
ok: [callisto-e1] => {
  "msg": "deploy_container_engine=False host=callisto-e1, etcd=host"
}
```